src/lib.rs

```diff
@@ -237,7 +237,7 @@ pub mod pipeline {
         fn matches(pattern: &lang::Search, raw: &str) -> bool {
             match *pattern {
                 lang::Search::MatchAll => true,
-                lang::Search::MatchFilter(ref filter) => raw.contains(filter),
+                lang::Search::MatchFilter(ref keyword) => keyword.to_regex().is_match(raw)
```

Comment:
I'm curious if this has a nontrivial impact on pipeline performance (which isn't a blocker, I would just call it out in the docs to use grep + `*` if you wanted to maximize perf). You can test performance by doing something like:
`yes | head -n 20000000 | agrind 'yes | count'` -- I'm curious if it makes a difference

It is possible to implement the sumologic filter without using a regex (I wrote the high performance matcher sumologic uses on streaming data when I was there) but it probably isn't worth it for angle grinder.

Reply:
👍 I read some more docs -- the reason why `parse` is so much slower using regex is the use of `capture_iterator` vs. `match`. The regex crate is highly optimized for `match`, but to actually build capture groups is has to do a bunch more work